format mismatch warnings

```
/home/joerg/src/avrdude/src/dfu.c:398:5: warning: format specifies type 'unsigned short' but the argument has type 'int' [-Wformat]
    ((unsigned short) dfu->dev_desc.bcdDevice >> 8) & 0xFF,
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/joerg/src/avrdude/src/avrdude.h:59:97: note: expanded from macro 'msg_info'
#define msg_info(...)       avrdude_message2(stderr, __LINE__, __FILE__, __func__, 0, MSG_INFO, __VA_ARGS__)
                                                                                                ^~~~~~~~~~~
/home/joerg/src/avrdude/src/dfu.c:399:5: warning: format specifies type 'unsigned short' but the argument has type 'int' [-Wformat]
    ((unsigned short) dfu->dev_desc.bcdDevice >> 4) & 0xF,
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/joerg/src/avrdude/src/avrdude.h:59:97: note: expanded from macro 'msg_info'
#define msg_info(...)       avrdude_message2(stderr, __LINE__, __FILE__, __func__, 0, MSG_INFO, __VA_ARGS__)
                                                                                                ^~~~~~~~~~~
/home/joerg/src/avrdude/src/dfu.c:400:5: warning: format specifies type 'unsigned short' but the argument has type 'int' [-Wformat]
    ((unsigned short) dfu->dev_desc.bcdDevice >> 0) & 0xF);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/joerg/src/avrdude/src/avrdude.h:59:97: note: expanded from macro 'msg_info'
#define msg_info(...)       avrdude_message2(stderr, __LINE__, __FILE__, __func__, 0, MSG_INFO, __VA_ARGS__)
…
/home/joerg/src/avrdude/src/flip2.c:559:5: warning: format specifies type 'unsigned short' but the argument has type 'int' [-Wformat]
    ((unsigned short) flip2->boot_ver >> 4) & 0xF,
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/joerg/src/avrdude/src/avrdude.h:59:97: note: expanded from macro 'msg_info'
#define msg_info(...)       avrdude_message2(stderr, __LINE__, __FILE__, __func__, 0, MSG_INFO, __VA_ARGS__)
                                                                                                ^~~~~~~~~~~
/home/joerg/src/avrdude/src/flip2.c:560:5: warning: format specifies type 'unsigned short' but the argument has type 'int' [-Wformat]
    ((unsigned short) flip2->boot_ver >> 0) & 0xF);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/joerg/src/avrdude/src/avrdude.h:59:97: note: expanded from macro 'msg_info'
#define msg_info(...)       avrdude_message2(stderr, __LINE__, __FILE__, __func__, 0, MSG_INFO, __VA_ARGS__)
                                                                                                ^~~~~~~~~~~
/home/joerg/src/avrdude/src/flip2.c:781:70: warning: format specifies type 'unsigned short' but the argument has type 'int' [-Wformat]
      pmsg_error("address out of range [0x%04hX,0x%04hX]\n", offset, offset+size-1);
                                                  ~~~~~              ^~~~~~~~~~~~~
                                                  %04X
/home/joerg/src/avrdude/src/avrdude.h:67:176: note: expanded from macro 'pmsg_error'
#define pmsg_error(...)     avrdude_message2(stderr, __LINE__, __FILE__, __func__, MSG2_PROGNAME|MSG2_FUNCTION|MSG2_FILELINE|MSG2_TYPE|MSG2_FLUSH|MSG2_LEFT_MARGIN, MSG_ERROR, __VA_ARGS__)
                                                                                                                                                                               ^~~~~~~~~~~
/home/joerg/src/avrdude/src/flip2.c:839:70: warning: format specifies type 'unsigned short' but the argument has type 'int' [-Wformat]
      pmsg_error("address out of range [0x%04hX,0x%04hX]\n", offset, offset+size-1);
                                                  ~~~~~              ^~~~~~~~~~~~~
                                                  %04X
/home/joerg/src/avrdude/src/avrdude.h:67:176: note: expanded from macro 'pmsg_error'
#define pmsg_error(...)     avrdude_message2(stderr, __LINE__, __FILE__, __func__, MSG2_PROGNAME|MSG2_FUNCTION|MSG2_FILELINE|MSG2_TYPE|MSG2_FLUSH|MSG2_LEFT_MARGIN, MSG_ERROR, __VA_ARGS__)
                                                                                                                                                                               ^~~~~~~~~~~
```
There would at least be an `(unsigned short)` typecast missing.

Anyway, I wonder what's the point of forcing all that into unsigned short, rather than keeping it simply `unsigned`, and remove all the `h` format modificators?

--- a/src/dfu.c
+++ b/src/dfu.c
@@ -394,10 +394,10 @@ void dfu_show_info(struct dfu_dev *dfu)
     msg_info("    USB Product         : 0x%04hX\n",
       (unsigned short) dfu->dev_desc.idProduct);
 
-  msg_info("    USB Release         : %hu.%hu.%hu\n",
-    ((unsigned short) dfu->dev_desc.bcdDevice >> 8) & 0xFF,
-    ((unsigned short) dfu->dev_desc.bcdDevice >> 4) & 0xF,
-    ((unsigned short) dfu->dev_desc.bcdDevice >> 0) & 0xF);
+  msg_info("    USB Release         : %u.%u.%u\n",
+    (dfu->dev_desc.bcdDevice >> 8) & 0xFF,
+    (dfu->dev_desc.bcdDevice >> 4) & 0xF,
+    (dfu->dev_desc.bcdDevice >> 0) & 0xF);
 
   if (dfu->serno_str != NULL)
     msg_info("    USB Serial No       : %s\n", dfu->serno_str);

--- a/src/flip2.c
+++ b/src/flip2.c
@@ -555,9 +555,9 @@ static void flip2_show_info(struct flip2 *flip2) {
       (char) (flip2->part_rev / 26 - 1 + 'A'),
       (char) (flip2->part_rev % 26 + 'A'));
 
-  msg_info("    Bootloader version  : 2.%hu.%hu\n",
-    ((unsigned short) flip2->boot_ver >> 4) & 0xF,
-    ((unsigned short) flip2->boot_ver >> 0) & 0xF);
+  msg_info("    Bootloader version  : 2.%u.%u\n",
+    (flip2->boot_ver >> 4) & 0xF,
+    (flip2->boot_ver >> 0) & 0xF);
 
   msg_info("    USB max packet size : %hu\n",
     (unsigned short) flip2->dfu->dev_desc.bMaxPacketSize0);
@@ -778,7 +778,7 @@ flip2_read_max1k_status:
     if (status.bStatus == ((FLIP2_STATUS_OUTOFRANGE >> 8) & 0xFF) &&
         status.bState == ((FLIP2_STATUS_OUTOFRANGE >> 0) & 0xFF))
     {
-      pmsg_error("address out of range [0x%04hX,0x%04hX]\n", offset, offset+size-1);
+      pmsg_error("address out of range [0x%04X,0x%04X]\n", offset, (offset+size-1) & 0xffff);
     } else
       pmsg_error("DFU status %s\n", flip2_status_str(&status));
     dfu_clrstatus(dfu);
@@ -836,7 +836,7 @@ static int flip2_write_max1k(struct dfu_dev *dfu,
     if (status.bStatus == ((FLIP2_STATUS_OUTOFRANGE >> 8) & 0xFF) &&
         status.bState == ((FLIP2_STATUS_OUTOFRANGE >> 0) & 0xFF))
     {
-      pmsg_error("address out of range [0x%04hX,0x%04hX]\n", offset, offset+size-1);
+      pmsg_error("address out of range [0x%04X,0x%04X]\n", offset, (offset+size-1) & 0xffff);
     } else
       pmsg_error("DFU status %s\n", flip2_status_str(&status));
     dfu_clrstatus(dfu);

--- a/src/updi_link.c
+++ b/src/updi_link.c
@@ -90,7 +90,7 @@ static int updi_physical_send(const PROGRAMMER *pgm, unsigned char *buf, size_t 
   size_t i;
   int rv;
 
-  pmsg_debug("sending %lu bytes [", len);
+  pmsg_debug("sending %lu bytes [", (unsigned long) len);
   for (i=0; i<len; i++) {
     msg_debug("0x%02x", buf[i]);
     if (i<len-1) {
@@ -114,7 +114,7 @@ static int updi_physical_recv(const PROGRAMMER *pgm, unsigned char *buf, size_t 
     return -1;
   }
 
-  pmsg_debug("received %lu bytes [", len);
+  pmsg_debug("received %lu bytes [", (unsigned long) len);
   for (i=0; i<len; i++) {
     msg_debug("0x%02x", buf[i]);
     if (i<len-1) {

